activemodel/lib/active_model/naming.rb

```diff
@@ -130,7 +130,7 @@ class Name
     #
     # Equivalent to +to_s+.
     delegate :==, :===, :<=>, :=~, :"!~", :eql?, :to_s,
-             :to_str, to: :name
+             :to_str, :to_json, to: :name
```

Comment:
I think you should be delegating `as_json` which returns a JSON-convertible Ruby object such as a String instance. `to_json` returns the actual escape JSON, e.g. it should return `"\"User\""` rather than `"User"`